### PR TITLE
Feature/queen synchro

### PR DESCRIPTION
**Pull request by davdarras**

### CODE REVIEW

File: src/main/java/fr/insee/publicenemy/api/infrastructure/questionnaire/entity/QuestionnaireEntity.java [hunk 1]
```diff
@@ -1,81 +1,175 @@
...
     }
 
+    /**
+     * Update synchronisation state for the questionnaire entity
+     * @param questionnaire with synchronisation state
+     */
+    public void updateState(@NotNull Questionnaire questionnaire) {
+        this.isSynchronized = questionnaire.isSynchronized();
+        questionnaire.getQuestionnaireModes().stream()
+                .forEach(questionnaireMode -> {
```
Comment (nsenave) [inline src/main/java/fr/insee/publicenemy/api/infrastructure/questionnaire/entity/QuestionnaireEntity.java:121]: Maybe `questionnaire.getQuestionnaireModes().forEach(...)`

File: src/main/java/fr/insee/publicenemy/api/infrastructure/questionnaire/entity/QuestionnaireEntity.java [hunk 2]
```diff
@@ -1,81 +1,175 @@
...
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        QuestionnaireEntity that = (QuestionnaireEntity) o;
+        return Objects.equals(id, that.id) && Objects.equals(poguesId, that.poguesId)
+                && Objects.equals(label, that.label) && context == that.context
+                && Objects.equals(modeEntities, that.modeEntities)
+                && Objects.equals(creationDate, that.creationDate)
+                && Objects.equals(updatedDate, that.updatedDate)
+                && Objects.equals(surveyUnitData, that.surveyUnitData)
```
Comment (nsenave) [inline src/main/java/fr/insee/publicenemy/api/infrastructure/questionnaire/entity/QuestionnaireEntity.java:152]: Warning that seem pertinent to me: 

> Arrays comparison should probably be done using 'Arrays.equals()'

File: src/main/java/fr/insee/publicenemy/api/infrastructure/questionnaire/entity/QuestionnaireModeEntity.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    @Id
+    @ManyToOne
+    @JoinColumn(name = "questionnaire_id")
+    private QuestionnaireEntity questionnaire;
+
+    @Column(name="state")
+    private String synchronisationState;
+
+    /**
+     * Constructorer
```
Comment (nsenave) [inline src/main/java/fr/insee/publicenemy/api/infrastructure/questionnaire/entity/QuestionnaireModeEntity.java:35]: (typo 🙃 )

File: src/main/java/fr/insee/publicenemy/api/infrastructure/questionnaire/entity/QuestionnaireModeEntity.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+     */
+    public static List<QuestionnaireMode> toModel(List<QuestionnaireModeEntity> modeEntities) {
+        return modeEntities.stream().map(QuestionnaireModeEntity::toModel).toList();
+    }
+
+    /**
+     * Permits to create a mode entity before saving it to persistence unit
+     * @param questionnaireEntity the questionnaire entity to link to the mode entity
+     * @param questionnaireMode application mode model
+     * @return
```
Comment (nsenave) [inline src/main/java/fr/insee/publicenemy/api/infrastructure/questionnaire/entity/QuestionnaireModeEntity.java:68]: missing description of return

File: src/test/java/fr/insee/publicenemy/api/application/usecase/QueenSynchronizationUseCaseTest.java [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+        when(questionnaire.getQuestionnaireModes()).thenReturn(questionnaireModes);
+        modes.stream().forEach(mode -> {
+            when(ddiUseCase.getJsonLunatic(ddi, context, mode)).thenReturn(map.get(mode));
+        });
+
+        queenUseCase.synchronizeCreate(ddi, context, questionnaire);
+        modes.stream().forEach(mode -> {
+            verify(queenServicePort).createQuestionnaireModel(any(), eq(ddi), eq(map.get(mode)));
+        });
+    }
```
Comment (nsenave) [inline src/test/java/fr/insee/publicenemy/api/application/usecase/QueenSynchronizationUseCaseTest.java:101]: (syntax)

```java
        modes.forEach(mode -> 
            when(ddiUseCase.getJsonLunatic(ddi, context, mode)).thenReturn(map.get(mode))
        );

        ...

        modes.forEach(mode -> 
            verify(queenServicePort).createQuestionnaireModel(any(), eq(ddi), eq(map.get(mode)))
        );
```

**Review by nsenave — COMMENTED** [review]
With some details, this seems good to me

**Comment by davdarras** [comment]
Thanks for the review !

**Pull request merged**